spec/unit/provider/exec/powershell_spec.rb

```diff
@@ -115,6 +116,19 @@
       }
     MANIFEST
     }
+    let(:tmpdir) { Dir.mktmpdir('statetmp').encode!(Encoding::UTF_8) }
+
+    before :each do
+      # a statedir setting must now exist per the new transactionstore code
+      # introduced in Puppet 4.6 for corrective changes, as a new YAML file
+      # called transactionstore.yaml will be written under this path
+      # which defaults to c:\dev\null when not set on Windows
+      Puppet[:statedir] = tmpdir
+    end
+
+    after :each do
+      FileUtils.rm_rf(tmpdir)
+    end
```

Comment:
On second thought, we probably want `/dev/null` behavior for the sake of specs.

PSH should probably be updated to mock out the transaction store code, which most implementers will not need for the sake of writing tests like this / against `puppet apply`